File: 11_phonetic_algorithms/tests/test_soundex.dart
import 'package:test/test.dart';
import 'package:soundex/soundex.dart';

void main() {
  group('Soundex', () {
    test('standard cases', () {
      expect(soundex("Robert"), equals("R163"));
      expect(soundex("Rupert"), equals("R163"));
      expect(soundex("Smith"), equals("S530"));
      expect(soundex("Smyth"), equals("S530"));
    });

    test('double letters', () {
      expect(soundex("Gutierrez"), equals("G362"));
    });

    test('repeat output numbers', () {
      expect(soundex("Pfister"), equals("P236"));
      expect(soundex("Jackson"), equals("J250"));
      expect(soundex("Tymczak"), equals("T522"));
    });

    test('repeat consonant separators', () {
      expect(soundex("Pfister"), equals("P236"));
      expect(soundex("Ashcraft"), equals("A261"));
    });
  });
}

File: 11_phonetic_algorithms/tests/test_soundex.py
from src.soundex import soundex


def test_soundex_standard_cases():
    assert soundex("Robert") == "R163"
    assert soundex("Rupert") == "R163"
    assert soundex("Smith") == "S530"
    assert soundex("Smyth") == "S530"


def test_soundex_double_letters():
    assert soundex("Gutierrez") == "G362"


def test_soundex_repeat_output_numbers():
    assert soundex("Pfister") == "P236"
    assert soundex("Jackson") == "J250"
    assert soundex("Tymczak") == "T522"


def test_soundex_repeat_consonant_separators():
    assert soundex("Tymczak") == "T522"
    assert soundex("Ashcraft") == "A261"
